### PR TITLE
FeedbackDetailsFragment: made fragment independent on NavigationView.

### LINKED ISSUE #3841: FeedbackDetailsFragment flow is broken
Feedback Details flow is broken in case if it's out of `NavigationView`. It expects parent's view contains `feedbackFrameLayout` container

https://github.com/mapbox/mapbox-navigation-android/blob/356e9d150b2e1ca5ea5b9c7222bc222316d60d23/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/feedback/FeedbackDetailsFragment.kt#L226-L234

https://github.com/mapbox/mapbox-navigation-android/blob/356e9d150b2e1ca5ea5b9c7222bc222316d60d23/libnavigation-ui/src/main/res/layout/mapbox_navigation_view.xml#L2-L6

**cc** @LukasPaczos @Guardiola31337

### PR DESCRIPTION
**Pull request by RingerJK**
### Description
`FeedbackDetailsFragment` holds a resource(link `R.id.feedbackFrameLayout `) of `NaivgationView` so when it's used out of `NavigationView` throws exceptions like "Resource not found"

Closes #3841 

### Changelog

```
<changelog>Fixed Exception `Resource not found` when using `FeedbackDetailsFragment`.</changelog>
```

### CODE REVIEW

**Review by Guardiola31337 — COMMENTED** [review]
Took this for a quick spin and noticed that although https://github.com/mapbox/mapbox-navigation-android/issues/3828 leak is not popping-up we're recreating the form and can be fill it multiple times 👀 

2tap_feedback_flow_issue_1

2tap_feedback_flow_issue_2

2tap_feedback_flow_issue_3

Discussing internally with @RingerJK and after https://github.com/mapbox/mapbox-navigation-android/pull/3842/commits/ade0b5103c1c71f85ba5eb43808bce6cd5438900 not getting multiple pop-ups anymore but can reproduce https://github.com/mapbox/mapbox-navigation-android/issues/3828 leak

**Comment by Guardiola31337** [comment]
`ktlint` is ❌ 

```
> Task :libnavigation-ui:ktlint FAILED
/root/code/libnavigation-ui/src/main/java/com/******/navigation/ui/internal/feedback/FeedbackDetailsFragment.kt:19:1: Unused import
```

**Review by Guardiola31337 — APPROVED** [review]
Approving with known issues ✅

**Comment by LukasPaczos** [comment]
Shouldn't we rather defer to the implementation of the feedback flow listener to inflate the arrival fragment after the details are complete?

**Comment by RingerJK** [comment]
> Shouldn't we rather defer to the implementation of the feedback flow listener to inflate the arrival fragment after the details are complete?

What’re pros of inflating Details and Arrival Feedback views independently?

**Comment by LukasPaczos** [comment]
The developer could choose where the fragment should be inflated into and manage its lifecycle instead of us defaulting to the root, `content` layout. But I guess it's already possible by passing `false` for the automatic inflation and then inflating manually in the details flow result. A tiny bit confusing, but not a blocker in any way.

**Pull request merged**
